spec/models/publish_services_event_metadata_spec.rb

```diff
@@ -51,4 +53,87 @@
+        keyUse: 'Signing',
+        federationEntityType: 'RP',
+      }].to_json
+    }
+
+    after(:each) do
+      SpComponent.destroy_all
+      MsaComponent.destroy_all
+    end
+
+    context 'does not occur' do
+      require 'polling/dev_cert_status_updater'
+      let(:sp_signing_certificate) { create(:sp_signing_certificate) }
+      it 'updates certificate in_use_at' do
+
+        expect(Certificate.find_by_id(sp_signing_certificate.id).in_use_at).to be_nil
+        dev_cert_status_updater = DevCertStatusUpdater.new
+        allow(subject).to receive(:poll).with(scheduler: Polling::Scheduler.new, status_updater: dev_cert_status_updater, hub_api: HUB_CONFIG_API)
+        allow(dev_cert_status_updater).to receive(:update_hub_usage_status_for_cert).with(HUB_CONFIG_API, sp_signing_certificate)
+          .and_return(CertificateInUseEvent.create(certificate: sp_signing_certificate))
+
+        create(:assign_sp_component_to_service_event, service: service, sp_component_id: sp_signing_certificate.component.id)
+        expect(Certificate.find_by_id(sp_signing_certificate.id).in_use_at).not_to be_nil
+      end
+    end
+
+    context 'occurs' do
+      let(:msa_encryption_certificate) { create(:msa_encryption_certificate) }
+      let(:sp_signing_certificate) { create(:sp_signing_certificate) }
+      it 'updates certificate in_use_at' do
+        stub_signing_certificates_hub_request(
+          environment: sp_signing_certificate.component.environment,
+          entity_id: service.entity_id
+        )
+        .to_return(body: hub_response_for_signing_certificate)
+
+        expect(Certificate.find_by_id(sp_signing_certificate.id).in_use_at).to be_nil
+
+        create(:assign_sp_component_to_service_event, service: service, sp_component_id: sp_signing_certificate.component.id)
+        expect(Certificate.find_by_id(sp_signing_certificate.id).in_use_at).not_to be_nil
+      end
+
+      it 'polls until certificate is in use then updates in_use_at' do
+        stub_signing_certificates_hub_request(
+          environment: sp_signing_certificate.component.environment,
+          entity_id: service.entity_id
+        )
+        .to_return(status: 404)
+        .times(2).then
+        .to_return(status: 200, body: hub_response_for_signing_certificate)
+
+        expect(Certificate.find_by_id(sp_signing_certificate.id).in_use_at).to be_nil
+
+        create(:assign_sp_component_to_service_event, service: service, sp_component_id: sp_signing_certificate.component.id)
+
+        loop { break if Certificate.find_by_id(sp_signing_certificate.id).in_use_at }
+        expect(Certificate.find_by_id(sp_signing_certificate.id).in_use_at).not_to be_nil
+      end
+
+      it 'polls but certificate is not in use' do
+
+        stub_signing_certificates_hub_request(
+          environment: sp_signing_certificate.component.environment,
+          entity_id: service.entity_id
+        )
+        .to_return(status: 404)
+        .times(2).then
```

Comment:
shouldn't it be this, then you don't need the last `.to_return` since it's again returning 404
```suggestion
        .times(3)
```